doc/dao/Getting Started with Development Environment & Running Primero.md

```diff
@@ -0,0 +1,291 @@
+
+`nvm` is a version manager for node runtimes. It is similar to rbenv, but manages node rather than ruby.
+
+`nvm` has a convenience script which will install it automatically. It should also automatically update your shell startup file.
+
+```shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+```
+
+In order to start using `nvm`, you will need to close and reopen your terminal.
+
+To install the version of node needed by primero, run the following command:
+```shell
+nvm install --lts
+```
+## Installing Docker and docker-compose
+
+> [!NOTE] 
+> If you already have docker and docker-compose, skip this step.
+> 
+> Docker can be installed in a number of different ways. As we are primarily using docker as a convienent way of running a database locally, it doesn't matter how it is installed.
+
+Install docker using apt:
+```bash
+sudo apt install docker.io docker-compose
+```
+## Building the Containers
+
+```bash
+cd ./docker
+sudo ./build.sh postgres solr
+```
+## Running the Containers
+
+You should run this from inside the docker directory.
+```bash
+sudo ./compose.local.sh up -d postgres
+sudo ./compose.local.sh run solr make-primero-core.sh primero-test
+sudo ./compose.local.sh up -d solr
+```
+
+# Configuring Primero for Local Development
+
+Primero is partially configured with a number of yaml files. There are example versions of these files provided for local development. They need to be copied to the correct locations in order for Primero to function.
+
+Execute these from the root directory of the repository.
+```bash
+cp config/database.yml.development config/database.yml
+cp config/locales.yml.development config/locales.yml
+cp config/mailers.yml.development config/mailers.yml
+cp config/sunspot.yml.development config/sunspot.yml
+mkdir log
+```
+
+You will also need to install some system-wide dependencies required to build and run Primero.
+
+```bash
+sudo apt install libpq-dev imagemagick libsodium-dev p7zip
+```
+
+Execute the following to install Primero's ruby and node dependencies:
+
+```bash
+bundle install
+npm ci
+```
+
+You now need to create the two Postgres databases (the default database, and a database used for unit testing).
+
+```shell
+bundle exec rails db:create
+bundle exec rails db:migrate
+bundle exec rails db:seed
+
+RAILS_ENV=test bundle exec rails db:migrate
+```
+
+You also need to generate the translation files.
+
+```shell
+bundle exec rails primero:i18n_js
+```
+
+Finally, you need to set a number of environment variables which contain necessary secrets.
+
+The environment variables are:
+- PRIMERO_SECRET_KEY_BASE
+- DEVISE_SECRET_KEY
+- DEVISE_JWT_SECRET_KEY
+
+The following bash command will generate a secure secret for each of these and add them to your `~/.bashrc`:
+
+```bash
+for v in PRIMERO_SECRET_KEY_BASE DEVISE_SECRET_KEY DEVISE_JWT_SECRET_KEY;
+do echo "export ${v}=$(openssl rand -hex 16)" >> ~/.bashrc; 
+done;
+```
+
+# Running Unit Tests
```

Comment:
Food for thought later, but we need to make sure that the unit test and linter pipelines in Github are solidly automated and not flaky.

Reply:
I also updated the little travis CI badge at the top of the readme to point to the analogous GitHub actions badge.